be/src/util/runtime_profile.cpp

```diff
@@ -103,16 +103,17 @@ void RuntimeProfile::merge(RuntimeProfile* other) {
             dst_iter = _counter_map.find(src_iter->first);
 
             if (dst_iter == _counter_map.end()) {
-                _counter_map[src_iter->first] =
-                        _pool->add(new Counter(src_iter->second->type(), src_iter->second->value()));
+                _counter_map[src_iter->first] = std::make_pair(
+                        _pool->add(new Counter(src_iter->second.first->type(), src_iter->second.first->value())),
```

Comment:
When will it be released?

Reply:
It's lifecycle is managed by `ObjectPool`, so when query finished, `ObjectPool` will be released, as well as all the objects in the pool